mne_connectivity/decoding/tests/test_decomposition.py

```diff
@@ -234,11 +241,173 @@ def test_spectral_decomposition(method, mode):
+        info=data.info,
+        method=method,
+        indices=(seeds, targets),
+        mode=mode,
+        fmin=np.min(fbands[:-1]) - 1,
+        fmax=np.max(fbands[:-1]) + 1,
+        cwt_freqs=cwt_freqs_fit,
+        cwt_n_cycles=cwt_n_cycles,
+        rank=(len(data_sims), len(data_sims)),
+    )
+    epochs_transformed = decomp_class.fit_transform(X=data.get_data())
+
+    # Compute connectivity
+    con = spectral_connectivity_epochs(
+        epochs_transformed,
+        method=bivariate_method,
+        indices=decomp_class.get_transformed_indices(),
+        sfreq=data.info["sfreq"],
+        mode=mode,
+        fmin=fstart,
+        fmax=fend,
+        cwt_freqs=cwt_freqs_con,
+        cwt_n_cycles=cwt_n_cycles,
+    )
+
+    # Check results
+    comp_i = 0
+    conn_scores = np.zeros((len(data_sims),))
+    for fband, snr, angle, dominant_chan in zip(
+        fbands[:-1], snrs[:-1], angles[:-1], dominant_chans[:-1]
+    ):
+        # find freqs. of interest
+        freqs = np.array(con.freqs)
+        freqs_conn = (freqs >= fband[0]) & (freqs <= fband[1])
+        freqs_noise = (freqs > fband[1]) | (freqs < fband[0])
+
+        # determine what is > noise based on SNR of simulated interaction
+        con_diff = snr / 2
+        if method == "mic":
+            con_diff *= 0.5  # connectivity strength lost when projecting to imag. axis
+        if mode in ["fourier", "cwt_morlet"]:
+            con_diff *= 0.5  # noise floor higher for Fourier and Morlet modes
+
+        # Check connectivity of genuine interactions greater than noise
+        assert (
+            np.abs(con.get_data()[comp_i, freqs_conn]).mean()
+            > np.abs(con.get_data()[comp_i, freqs_noise]).mean() + con_diff
+        )  # check con. > noise in same component
+        assert (
+            np.abs(con.get_data()[comp_i, freqs_conn]).mean()
+            > np.abs(con.get_data()[-1, :]).mean() + con_diff
+        )  # check con. > noise in noise component
+
+        if method == "cacoh":
+            # Check filters extract components at correct angles
+            assert_allclose(
+                np.angle(con.get_data()[comp_i, freqs_conn], deg=True).mean(),
+                angle,
+                atol=10,
+            )
+
+        if mode in ["multitaper", "fourier"]:  # less stability for Morlet mode
+            # Check patterns reflect spatial distribution of activity
+            nondom_channs = np.ones(decomp_class.patterns_[0].shape[1], dtype=bool)
+            nondom_channs[dominant_chan] = False
+            assert np.all(
+                np.abs(decomp_class.patterns_[0][comp_i, dominant_chan])
+                > np.abs(decomp_class.patterns_[0][comp_i, nondom_channs])
+            )  # check for seeds
+            assert np.all(
+                np.abs(decomp_class.patterns_[1][comp_i, dominant_chan])
+                > np.abs(decomp_class.patterns_[1][comp_i, nondom_channs])
+            )  # check for targets
```

Comment:
prefer `assert_array_less`, the error message is more informative
```suggestion
            assert_array_less(
                np.abs(decomp_class.patterns_[1][comp_i, nondom_channs]),
                np.abs(decomp_class.patterns_[1][comp_i, dominant_chan]),
                err_msg="targets",
            )  # check for targets
```
same elsewhere

Reply:
Sorry, keep forgetting this. Should be all instances changed in 69d9a3db1cc0b56633daa3ca1bbfb87ea5195530.